README-wsl2-windows.md

```diff
@@ -0,0 +1,118 @@
+# Using WSL2 for esp-idf-template on Windows
+
+This doc will help you set up and use WSL2 (Windows Subsystem for Linux 2) for development using the esp-idf-template on Windows. 
+This method avoids [path length issues]([Error: Too long output directory · Issue #252 · esp-rs/esp-idf-sys (github.com)](https://github.com/esp-rs/esp-idf-sys/issues/252)) and provides a smoother development experience.
```

Comment:
Instead of "This method avoids" perhaps something like "Using WSL2 does **not** exhibit the [path length issues..."

Also, "and provides a smoother development experience" is subjective. We should either remove it or if we want to cite an extra concrete technical benefit - perhaps something like "; furthermore, using WSL2 reduces the waiting time between command line cargo invocations and Rust Analyzer operating on the same projects" - as this is a real problem on native Windows (perhaps, due to the different filesystem semantics in native Win32).